jenkins/Hono-Website-Pipeline.groovy

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env groovy
+
+/*******************************************************************************
+ * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+/**
+ * Builds Hono's web site using Hugo every night (between 3 and 4 AM) and publishes to Eclipse.
+ *
+ */
+
+node {
+    def utils = evaluate readTrusted("jenkins/Hono-PipelineUtils.groovy")
+    properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '3')),
+                pipelineTriggers([cron('TZ=Europe/Berlin \n # every night between 3 and 4 AM \n H 3 * * *')])])
+    try {
+        deleteDir()
+        build()
+        currentBuild.result = 'SUCCESS'
+    } catch (err) {
+        currentBuild.result = 'FAILURE'
+        echo "Error: ${err}"
+    }
+    finally {
+        echo "Build status: ${currentBuild.result}"
+        utils.notifyBuildStatus()
+    }
+}
+
+/**
+ * Build stages for Hono's web site pipeline using Hugo
+ *
+ */
+def build() {
+
+    stage('Checkout to sub-directory "hono"') {
+        echo "Check out master to subdirectory [hono]"
+        checkout([$class                           : 'GitSCM',
```

Comment:
Hmmm, I think we do not need this block at all and can instead simply do an anonymous checkout using a shell command as in the next stage. We do not change the content of the GitHub repo, thus IMHO we do not need to authenticate ...